visitor.go

```diff
@@ -38,12 +39,18 @@ func (vis *visitor) mustVisit(w io.Writer, v Value) {
 	}
 	defer vis.leave(v)
 
+	cw := count.NewWriter(w)
+
 	for _, f := range vis.filters {
-		if n := must.Must(f(w, v)); n > 0 {
-			return
+		if err := f(cw, v, vis.visit); err != nil {
+			panic(must.PanicSentinel{Cause: err})
 		}
 	}
 
+	if cw.Count() > 0 {
```

Comment:
This check needs to be inside the loop above, otherwise multiple filters could be engaged for a single value.